clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/view/OpenstackSecurityGroupProvider.groovy

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.provider.view
+
+import com.netflix.spinnaker.clouddriver.model.SecurityGroupProvider
+import com.netflix.spinnaker.clouddriver.openstack.OpenstackCloudProvider
+import com.netflix.spinnaker.clouddriver.openstack.model.OpenstackSecurityGroup
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackNamedAccountCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Slf4j
+@Component
+class OpenstackSecurityGroupProvider implements SecurityGroupProvider<OpenstackSecurityGroup> {
+
+  final AccountCredentialsProvider accountCredentialsProvider
+
+  @Autowired
+  OpenstackSecurityGroupProvider(AccountCredentialsProvider accountCredentialsProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider
+  }
+
+  Set<OpenstackSecurityGroup> getSecurityGroups() {
+    // TODO This is not thread safe and not cached
+    // This just proves the provider is working once it has a data set.
+    Set<OpenstackSecurityGroup> securityGroups = []
+    accountCredentialsProvider.all.each {account ->
+      if (account instanceof OpenstackNamedAccountCredentials) {
+        def provider = ((OpenstackNamedAccountCredentials) account).credentials.provider
+
+        provider.allRegions.each { region ->
```

Comment:
I agree, though this is just a place holder till the security groups caching is in place. Then this provider will just grab the data from the cache.